The request randomly generates a wiki about one character in the univ…

opopovych opened the pull request:
…erse the animated series Rick & Morty.

The request takes a string as an argument, and returns a list of all characters whose name contains the search string

opopovych committed:
The request randomly generates a wiki about one character in the universe the animated series Rick & Morty.
The request takes a string as an argument, and returns a list of all characters whose name contains the search string

koval-kn commented on src/main/java/mate/academy/rickandmorty/service/CharacterService.java:
```diff
@@ -0,0 +1,51 @@
+    public List<Character> searchCharactersByName(String searchQuery) {
+        return characterRepository.findByNameContaining(searchQuery);
+    }
+}
```
this should be interface and implementation

koval-kn commented on src/main/java/mate/academy/rickandmorty/service/DataLoadOnStartup.java:
```diff
@@ -0,0 +1,39 @@
+import org.springframework.stereotype.Component;
+
+@Component
+public class DataLoadOnStartup {
```
more suitable name
```suggestion
public class RickAndMortyClient {
```

koval-kn commented on src/main/java/mate/academy/rickandmorty/service/DataLoadOnStartup.java:
```diff
@@ -0,0 +1,39 @@
+        this.objectMapper = objectMapper;
+    }
+
+    public List<CharacterResultDto> getDataByApi() {
```
```suggestion
    public List<CharacterResultDto> getCharacters() {
```

koval-kn commented on src/main/java/mate/academy/rickandmorty/service/DataLoadOnStartup.java:
```diff
@@ -0,0 +1,39 @@
+    }
+
+    public List<CharacterResultDto> getDataByApi() {
+        HttpClient httpClient = HttpClient.newHttpClient();
```
would be nice to make it class-level `private final` field and create Spring bean for it, so we can autowire it

koval-kn commented on src/main/java/mate/academy/rickandmorty/controller/CharacterController.java:
```diff
@@ -0,0 +1,35 @@
+        return randomCharacter.toString();
+    }
+
+    @GetMapping("/naming")
```
```suggestion
    @GetMapping("/search")
```

koval-kn commented on src/main/java/mate/academy/rickandmorty/controller/CharacterController.java:
```diff
@@ -0,0 +1,35 @@
+    @Operation(summary = "Get all character which contain arguments",
+            description = "Get all character which contain arguments")
+    public String getCharactersContaining(@RequestParam String args) {
+        List<Character> characters = characterService.searchCharactersByName(args);
```
```suggestion
    public String searchCharactersByName(@RequestParam String nameContains) {
        List<Character> characters = characterService.searchCharactersByName(nameContains);
```

koval-kn reviewed (changes requested):
almost good

opopovych committed:
Created interface CharacterService.java
changed method's name to more suitable
HttpClient changed as private final field and initialized through constructor.

opopovych committed:
Recalled endpoint
created mapper
remove deprecated property and corrected all bugs

opopovych committed:
Created interface CharacterService.java
changed method's name to more suitable
HttpClient changed as private final field and initialized through constructor.

sokimaaa commented on res.json:
remove redundant files
edit gitignore to predict adding the redundant files

sokimaaa commented on src/main/java/mate/academy/rickandmorty/controller/CharacterController.java:
```diff
@@ -0,0 +1,35 @@
+public class CharacterController {
+    private final CharacterServiceImpl characterService;
+
+    @GetMapping("/random")
```
REST URI convnetion violations 

recall the convention and fix endpoints.

sokimaaa commented on src/main/java/mate/academy/rickandmorty/service/CharacterServiceImpl.java:
```diff
@@ -0,0 +1,55 @@
+    }
+
+    @Override
+    public Character convertDtoToCharacter(CharacterResultDto characterResultDto) {
+        Character character = new Character();
+        character.setExternalId(characterResultDto.getExternalId());
+        character.setName(characterResultDto.getName());
+        character.setStatus(characterResultDto.getStatus());
+        character.setGender(characterResultDto.getGender());
+        return character;
+    }
```
SRP Violation. 

Extract to diff class that responsible for mapping. 
consider using mapstruct. 
```suggestion
```

sokimaaa commented on src/main/java/mate/academy/rickandmorty/service/CharacterServiceImpl.java:
```diff
@@ -0,0 +1,55 @@
+    @Override
+    public Character getRandomCharacter() {
+        List<Character> characters = characterRepository.findAll();
+        Random random = new Random();
```
We are using the spring, so instantiation of service classes via `new` is bad practice. 

Make a bean of random and autowire it. 
```suggestion
        Random random = new Random();
```

sokimaaa commented on src/main/resources/application.properties:
```diff
@@ -1 +1,6 @@
-
+rickandmortyapi.base-url=https://rickandmortyapi.com/api/character
+spring.datasource.url=jdbc:h2:mem:testdb
+spring.datasource.driverClassName=org.h2.Driver
```
is `spring.datasource.driverClassName` required property?
is `spring.jpa.database-platform=org.hibernate.dialect.H2Dialect` required property? 

I suppose, they are deprecated.

opopovych committed:
Recalled endpoint
created mapper
remove deprecated property and corrected all bugs

opopovych committed:
updated gitignore
